harness/assert.js

```diff
@@ -62,3 +62,64 @@ assert.throws = function (expectedErrorConstructor, func) {
 
     $ERROR('Expected a ' + expectedErrorConstructor.name + ' to be thrown but no exception was thrown at all');
 };
+
+(function() {
+  function classOf(object) {
+    // Argument must not be null or undefined.
+    var string = Object.prototype.toString.call(object);
+    // String has format [object <ClassName>].
+    return string.substring(8, string.length - 1);
+  }
+
+  function deepObjectEquals(a, b) {
+    var aProps = Object.keys(a);
```

Comment:
@bterlson I think it would be better to explicitly build this array (and the one following it) with a for..in loop and "own" property check. Do you agree?